coalib/misc/DictUtilities.py

```diff
@@ -11,26 +11,31 @@ def inverse_dicts(*dicts):
     No order is preserved.
 
     :param dicts: The dictionaries to invert.
+    :type: dict
     :return:      The inversed dictionary which merges all dictionaries into
                   one.
+    :rtype: defaultdict of lists
     """
-    inverse = {}
+    inverse = defaultdict(list)
 
     for dictionary in dicts:
         for key, value in dictionary.items():
             if isinstance(value, Iterable):
                 for item in value:
-                    add_pair_to_dict(item, key, inverse)
+                    inverse[item].append(key)
             else:
-                add_pair_to_dict(value, key, inverse)
-
+                inverse[value].append(key)
     return inverse
 
 
 def add_pair_to_dict(key, value, dictionary):
```

Comment:
FWIW I think we should remove it - it just adds to code baggage. It's specific to this file and it's not used anywhere.